akka-actor/src/main/scala/akka/japi/JavaAPI.scala

```diff
@@ -258,7 +258,7 @@ object Util {
         if (i.hasNext) {
           val builder = new immutable.VectorBuilder[T]
 
-          do { builder += i.next() } while (i.hasNext)
+          while ({ builder += i.next(); i.hasNext }) ()
```

Comment:
Kind of funky to do the mutation inside the guard, but I guess it was optimized to not call hasNext an extra time, so I think that is fine.